inc/blocks/namespace.php

```diff
@@ -157,6 +158,84 @@ function on_save_post( int $post_ID, WP_Post $post, bool $update ) : void {
 	}
 }
 
+/**
+ * Synchronise any XBs on the widget page with a shadow post type.
+ *
+ * @param instance $instance Widget Object.
+ * @param new_instance $new_instance Widget Object.
+ * @return object
+ */
+function on_widgets_save( $instance, $new_instance ) {
+
+	// Scan for XBs in the post content.
+	$blocks = parse_blocks( $new_instance['content'] );
+	$xbs = find_xbs( $blocks );
+
+	// Find referenced XBs.
+	$existing_posts = new WP_Query( [
+		'post_type' => POST_TYPE,
+		'post_status' => 'any',
+		'post_name__in' => array_filter( array_map( function ( array $xb ) : string {
+			return $xb['attrs']['clientId'];
+		}, $xbs ) ),
+		'posts_per_page' => 100,
+		'no_found_rows' => true,
+	] );
+
+	// Update or create XB posts.
+	foreach ( $xbs as $index => $xb ) {
+		if ( empty( $xb['attrs']['clientId'] ) ) {
+			continue;
+		}
+
+		// Extract existing post if found.
+		$posts = wp_list_filter( $existing_posts->posts, [
+			'post_name' => $xb['attrs']['clientId'],
+		] );
+		$posts = array_values( $posts ); // Reset keys.
+
+		// Generate a default using the current post title and instance number in the content.
+		$default_title = sprintf( '(XB %s)', $index + 1 );
```

Comment:
Im finding that there is only a single item so the  index is always going to be 1. 

Originally the code was : `$default_title = sprintf( '%s (XB %s)', $post->post_title, $index + 1 );`

However there is no `$post->post_title`, hence the change.

( I need to adjust the comment more appropriately for the title once a solution is possibly implemented here )

Reply:
So each block counts as a widget? Can’t have multiple blocks in a widget? If so then it’s fine to omit that, you might even be able to fall back to or reference the widget title from the instance data

Reply:
This will always be "(XB 1)" which isn't a super helpful title.

If you update the action hook to pass the 3rd parameter which is the `WP_Widget` object, that has a `number` property that could be used in place of `$index + 1`, e.g. `$widget->number`. Don't forget to update how many arguments the hook receives too.

I would probably also check the block type, plus we know the context here so we can generate default titles like:

"Widget A/B Test (XB 1)"

Ideally in the JS we would have the same kind of fallback placeholder title if possible, you need to look at the context, maybe if `currentPost` is null?